lib/luaT/luaT.c

```diff
@@ -471,53 +471,136 @@ void luaT_registeratname(lua_State *L, const struct luaL_Reg *methods, const cha
 }
 
 
-/* utility functions */
-const char *luaT_classrootname(const char *tname)
+/* returns the name of the class itself (sans nesting) */
+const char* luaT_classrootname(const char *tname)
 {
-  int i;
+  int idx;
   int sz = strlen(tname);
 
-  for(i = 0; i < sz; i++)
+  for(idx = sz-1; idx >= 0 ; idx--)
   {
-    if(tname[i] == '.')
-      return tname+i+1;
+    if(tname[idx] == '.')
+      return tname+idx+1;
   }
   return tname;
 }
 
-/* module_name must be a buffer at least as big as tname 
- * return true if the class is part of a module */
-int luaT_classmodulename(const char *tname, char *module_name)
+/* parent_name must be a buffer at least as big as tname.
+ * If class has a parent, returns true; and, sets
+ * parent name to that of full parent hierarchy (e.g.
+ * given class `A.b.c`, sets parent_name to `A.b`)
+ */
+int luaT_fullparentname(const char *tname, char *parent_name)
```

Comment:
I've added an alias of sorts s.t. calling `luaT_classmodulename` immediately redirects as a call to `luaT_fullparentname`. Please let me know if you'd like prefer I simply rename `luaT_fullparentname`; for the time being, I chose above recourse in that `luaT_fullparentname` seems to cohere better with `luaT_outerparentname` and `luaT_innerparentname`.